### ISSUE TITLE
Allow for-in initializers

### ISSUE THREAD

**Issue by bakkot** [tc39/test262-parser-tests#21]
[This test](https://github.com/tc39/test262-parser-tests/blob/master/fail/e3fbcf63d7e43ead.js) is misclassified as of https://github.com/tc39/ecma262/pull/614.

**Comment by littledan** [comment]
For-in initializers? I thought ES2015 got rid of those. What idiot would put those back in?

(I was!)

**Comment by leobalter** [comment]
@littledan if that's a joke it is going without a context and it's not fun. I recommend avoiding this tone.

**Comment by littledan** [comment]
Yes, it was a joke since I was the one to reintroduce it.

**Comment by jmdyck** [comment]
Yeah, but if the reader doesn't know that, it looks to them like you're calling someone else an idiot. And maybe they think "Ah, it's okay to insult people here."

If, in the same post, you had revealed that "the idiot" was you, then everyone would have realized you were being self-deprecating, and (in my opinion) the joke wouldn't have been a problem.

**Comment by littledan** [comment]
OK, edited the original comment to include the reveal.
